packages/dev/loaders/src/USD/usdFileLoader.ts

```diff
@@ -0,0 +1,223 @@
+     * @param meshesNames a string or array of strings of the mesh names that should be loaded from the file
+     * @param scene the scene the meshes should be added to
+     * @param data the USD data to load
+     * @param rootUrl root url to load from
+     * @param onProgress callback called while file is loading
+     * @param fileName Defines the name of the file to load
+     * @returns a promise containing the loaded meshes, particles, skeletons and animations
+     */
+    public async importMeshAsync(
+        meshesNames: any,
+        scene: Scene,
+        data: any,
+        rootUrl: string,
+        onProgress?: (event: ISceneLoaderProgressEvent) => void,
+        fileName?: string
+    ): Promise<ISceneLoaderAsyncResult> {
+        return this._parse(meshesNames, scene, data, rootUrl).then((meshes) => {
+            return {
+                meshes: meshes,
+                particleSystems: [],
+                skeletons: [],
+                animationGroups: [],
+                transformNodes: [],
+                geometries: [],
+                lights: [],
+                spriteManagers: [],
+            };
+        });
+    }
+    private static _UniqueResolveID = 1000;
+
+    private static _LoadScriptModuleAsync(scriptUrl: string, scriptId?: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            // Need a relay
+            let windowAsAny: any;
+            let windowString: string;
+
+            if (IsWindowObjectExist()) {
+                windowAsAny = window;
+                windowString = "window";
+            } else if (typeof self !== "undefined") {
+                windowAsAny = self;
+                windowString = "self";
+            } else {
+                reject(new Error("Cannot load script module outside of a window or a worker"));
+                return;
+            }
+
+            if (!windowAsAny._LoadScriptModuleResolve) {
+                windowAsAny._LoadScriptModuleResolve = {};
+            }
+            windowAsAny._LoadScriptModuleResolve[USDFileLoader._UniqueResolveID] = resolve;
+
+            scriptUrl += `
+                ${windowString}._LoadScriptModuleResolve[${USDFileLoader._UniqueResolveID}](returnedValue);
+                ${windowString}._LoadScriptModuleResolve[${USDFileLoader._UniqueResolveID}] = undefined;
+            `;
+            USDFileLoader._UniqueResolveID++;
+
+            Tools.LoadScript(
+                scriptUrl,
+                undefined,
+                (message, exception) => {
+                    reject(exception || new Error(message));
+                },
+                scriptId,
+                true
+            );
+        });
+    }
+
+    private _initializeTinyUSDZAsync(): Promise<void> {
+        return USDFileLoader._LoadScriptModuleAsync(
+            `
+            import Module from 'https://lighttransport.github.io/tinyusdz/tinyusdz.js';

```

Comment:
We should do that differently in the framework itself. A few notes:

* Can we host it on our own CDN?
* We should provide it as part of our `@babylonjs/loaders` package, if possible
* This should be overridable. And injectable - i.e. people should be able to host it themselves, or load it themselves and pass the package to the loader

Reply:
I've reworked the options so user can change module url.
My intent is to do a PR on tinyusdz repo with a new Github action job that will build the wasm and upload it as a npm. Then, use `https://unpkg.com` like CSG2 did.

Reply:
Sorry, didn't see the csg2 code. i'll resolve this, but we will need to discuss hosting it ourselves.